apps/api/src/account/account.controller.ts

```diff
@@ -34,7 +35,45 @@ export class AccountController {
     @AuthenticatedUser() user: KeycloakTokenParsed,
     @Body() data: UpdatePersonDto,
   ) {
-    return await this.accountService.updateUserProfile(user.sub as string, data)
+    let person: Person | null = null
+    try {
+      person = await this.accountService.updateUserProfile(user, data)
+    } catch (err) {
+      Logger.error(`Failed to update user with keycloakId ${user.sub}. Error is: ${err}`)
+      throw err
+    }
+
+    return {
+      user: person,
+    }
+  }
```

Comment:
The following will have the same effect but with less type definitions:

```suggestion
    try {
      const user = await this.accountService.updateUserProfile(user, data)
      return { user }
    } catch (err) {
      Logger.error(`Failed to update user with keycloakId ${user.sub}. Error is: ${err}`)
      throw err
    }
  }
```